sui_programmability/framework/sources/Governance/EpochRewardRecord.move

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+module Sui::EpochRewardRecord {
+    use Sui::ID::VersionedID;
+    use Sui::Transfer;
+    use Sui::TxContext::{Self, TxContext};
+
+    friend Sui::ValidatorSet;
+
+    /// EpochRewardRecord is an immutable record created per epoch per active validator.
+    /// Sufficient information is saved in the record so that delegators can claim
+    /// delegation rewards from past epochs, and for validators that may no longer be active.
+    /// TODO: For now we assume that validators don't charge an extra fee.
+    /// Delegation reward is simply proportional to to overall delegatiomn reward ratio
+    /// and the delegation amount.
+    struct EpochRewardRecord has key {
+        id: VersionedID,
+        epoch: u64,
+        computation_charge: u64,
+        total_stake: u64,
+        validator: address,
+    }
+
+    public(friend) fun create(
+        epoch: u64,
+        computation_charge: u64,
+        total_stake: u64,
+        validator: address,
+        ctx: &mut TxContext,
+    ) {
+        Transfer::freeze_object(EpochRewardRecord {
+            id: TxContext::new_id(ctx),
+            epoch,
+            computation_charge,
+            total_stake,
+            validator,
+        })
+    }
+
+    public fun get_epoch(self: &EpochRewardRecord): u64 {
```

Comment:
I believe it is common to name getters after properties:
```suggestion
    public fun epoch(self: &EpochRewardRecord): u64 {
```

And same applies below.